src/hooks/useHtmlPaste/index.ts

```diff
@@ -0,0 +1,129 @@
+import {useNavigation} from '@react-navigation/native';
+import ExpensiMark from 'expensify-common/lib/ExpensiMark';
+import {useCallback, useEffect} from 'react';
+import type UseHtmlPaste from './types';
+
+const useHtmlPaste: UseHtmlPaste = (textInputRef, checkComposerVisibility, isUnmountedOnBlur = true) => {
+    const navigation = useNavigation();
+
+    /**
+     * Set pasted text to clipboard
+     * @param {String} text
+     */
+    const paste = useCallback((text: string) => {
+        try {
+            document.execCommand('insertText', false, text);
+            // Pointer will go out of sight when a large paragraph is pasted on the web. Refocusing the input keeps the cursor in view.
+            textInputRef.current?.blur();
+            textInputRef.current?.focus();
+            // eslint-disable-next-line no-empty
+        } catch (e) {}
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    /**
+     * Manually place the pasted HTML into Composer
+     *
+     * @param {String} html - pasted HTML
+     */
+    const handlePastedHTML = useCallback(
+        (html: string) => {
+            const parser = new ExpensiMark();
+            paste(parser.htmlToMarkdown(html));
+        },
+        [paste],
+    );
+
+    /**
+     * Paste the plaintext content into Composer.
+     *
+     * @param {ClipboardEvent} event
+     */
+    const handlePastePlainText = useCallback(
+        (event: ClipboardEvent) => {
+            const plainText = event.clipboardData?.getData('text/plain');
+            if (plainText) {
+                paste(plainText);
+            }
+        },
+        [paste],
+    );
+
+    const handlePaste = useCallback(
+        (event: ClipboardEvent) => {
+            if (!textInputRef.current) {
+                return;
+            }
+            const isVisible = typeof checkComposerVisibility === 'function' && checkComposerVisibility();
+            const isFocused = textInputRef.current?.isFocused();
+
+            if (!isFocused && !isVisible) {
+                return;
+            }
+
+            if (textInputRef.current !== event.target) {
+                // To make sure the text input does not capture paste events from other inputs, we check where the event originated
+                // If it did originate in another input, we return early to prevent the text input from handling the paste
+                const target = event.target as HTMLInputElement;
+                const isTargetInput = (target && target.nodeName === 'INPUT') || target.nodeName === 'TEXTAREA' || target.contentEditable === 'true';
+
+                if (isTargetInput) {
+                    return;
+                }
```

Comment:
copy-pasta from here:

https://github.com/Expensify/App/blob/d98a49506519198e3b64c958a8d48d3cc48251af/src/components/Composer/index.tsx#L171-L176

Maybe move this into a util or something?